datasets/flwr_datasets/partitioner/non_iid_partitioner.py

```diff
@@ -0,0 +1,68 @@
+from typing import Dict
+import datasets
+from flwr_datasets.partitioner.natural_id_partitioner import NaturalIdPartitioner
+import numpy as np
+
+class NonIidPartitioner(NaturalIdPartitioner):
```

Comment:
The `NonIidPartitioner` seems too general. I'd propose `GroupedNaturalIdPartitioner` which seems what the partitioner does.